src/js/components/modals/AppModalComponent.jsx

```diff
@@ -335,14 +335,14 @@ var AppModalComponent = React.createClass({
             </StoreFormGroupComponent>
             <div className="row">
               <div className="col-sm-3">
-                <FormGroupComponent
-                    attribute="cpus"
+                <StoreFormGroupComponent
+                    errorMessage={this.getErrorMessage("cpus")}
+                    fieldId="cpus"
                     label="CPUs"
-                    model={model}
-                    errors={errors}
-                    validator={appValidator}>
-                  <input min="0" step="any" type="number" required />
-                </FormGroupComponent>
+                    value={state.fields.cpus || 0.1}
```

Comment:
@aldipower where should we store default values? Maybe we could have a 
`getDefaultValue(fieldId)` function in the AppModalComponent that looks like this:

```
getDefaultValue(fieldId) {
  if (this.state[fieldId] == null) {
    return this.props[fieldId]
  } else {
    return this.state[fieldId];
  }
},
```

Of course this assumes we pass all the default values down as properties from the Marathon component, which might make sense.